apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -340,7 +340,21 @@ private static boolean isIncluded(ElasticApmInstrumentation advice, CoreConfigur
         if (disabledInstrumentations.contains("incubating")) {
             disabledInstrumentations.add("experimental");
         }
-        return !isGroupDisabled(disabledInstrumentations, advice.getInstrumentationGroupNames()) && isInstrumentationEnabled(advice, coreConfiguration);
+        return isGroupEnabled(coreConfiguration.getEnabledInstrumentations(), advice.getInstrumentationGroupNames()) &&
+            !isGroupDisabled(disabledInstrumentations, advice.getInstrumentationGroupNames()) &&
+            isInstrumentationEnabled(advice, coreConfiguration);
+    }
+
+    private static boolean isGroupEnabled(Collection<String> enabledInstrumentations, Collection<String> instrumentationGroupNames) {
```

Comment:
Consider moving the logic to `CoreConfiguration`:

* `public isEnabled(Collection<String> instrumentationGroupNames)` (calls `isGroupDisabled` and `isGroupEnabled)
  Naming is tricky here as it clashes with `isGroupEnabled` but the semantics are different. Maybe you ave a good idea?
* `public isEnabled(String instrumentationGroupName)`
* `private isGroupDisabled(Collection<String> instrumentationGroupNames)`
* `private isGroupEnabled(Collection<String> instrumentationGroupNames)`

Reply:
Done